src/allmydata/test/test_system.py

```diff
@@ -1811,6 +1814,45 @@ def _check_lit_filenode_results(r):
         d.addCallback(_got_lit_filenode)
         return d
 
+    @async_to_deferred
+    async def test_upload_download_immutable_different_default_max_segment_size(self):
```

Comment:
Perhaps the integration test suite is better suited to this kind of thing?

Reply:
Hmhm. I was thinking I'd need to push the segment size through another N layers of abstraction but I guess with a config option then maybe not?